aten/src/ATen/native/Math.h

```diff
@@ -311,74 +291,19 @@ static inline double calc_digamma(double x) {
       8.33333333333333333333E-2,
   };
 
-  double y = 0;
+  T y = 0;
   if (x < 1.0e17) {
-    double z = 1.0 / (x * x);
+    T z = 1.0 / (x * x);
     y = z * polevl(z, A, 6);
   }
   return result + log(x) - (0.5 / x) - y;
 }
 
-/*
- * This function is derived from the implementation of the digamma function in the Cephes Math Library.
- * See note [3-Clause BSD License for the Cephes Math Library].
- */
-static inline float calc_digamma(float x) {
-  static float PSI_10 = 2.25175258906672110764f;
-  if (x == 0) {
-    return INFINITY;
-  }
-
-  int x_is_integer = x == floorf(x);
-  if (x < 0) {
```

Comment:
@ita9naiwa Thanks for contributing this change.
I am fine with template, but the float version is not the same as other types. Can you clarify on why you think you can merge float version into your generic template version? i think you should have a float specialization version for your generic template to adopt the difference in float version, ie `template<> calc_digamma<float>`, any thoughts?

Reply:
Yes, I agree with that. I jsut technically moved double version functions to template functions and removed float equivalents, but there are possible differences between float version functions and double version functions.

I'll check them and add specialized functions if needed.